pennylane/operation.py

```diff
@@ -285,24 +286,42 @@ def __init__(self, *args, wires=None, do_queue=True):
             assert self.grad_recipe is None, 'Gradient recipe is only used by the A method!'
 
         # apply the operation on the given wires
-        if isinstance(wires, int):
-            wires = [wires]
-
-        if self.num_wires != 0 and len(wires) != self.num_wires:
-            raise ValueError("{}: wrong number of wires. "
-                             "{} wires given, {} expected.".format(self.name, len(wires), self.num_wires))
+        if not isinstance(wires, Sequence):
+            self._wires = [wires]
+        else:
+            self._wires = wires
 
-        if len(set(wires)) != len(wires):
-            raise ValueError('{}: wires must be unique, got {}.'.format(self.name, wires))
+        if all([isinstance(w, int) for w in self._wires]):
+            # If all wires are integers (i.e., not Variable), check
+            # that they are valid for the given operation
+            self.check_wires(self._wires)
```

Comment:
We still perform wire validation on construction **if** they do not contain variables.